go/vm/geth/ct.go

```diff
@@ -202,7 +202,27 @@ func (i *callInterceptor) CallCode(env *geth_vm.EVM, me geth_vm.ContractRef, add
 }
 
 func (i *callInterceptor) DelegateCall(env *geth_vm.EVM, me geth_vm.ContractRef, addr geth_common.Address, data []byte, gas uint64) ([]byte, uint64, error) {
-	panic("not implemented")
+	kind := vm.DelegateCall
+	if i.static {
+		kind = vm.StaticCall
+	}
+
+	var vmValue vm.Value
+	i.stateDb.GetBalance(me.Address()).FillBytes(vmValue[:])
+	res, _ := i.context.Call(kind, vm.CallParameter{
+		Sender:    vm.Address(me.Address()),
+		Recipient: vm.Address(addr),
+		Value:     vmValue,
+		Input:     data,
+		Gas:       vm.Gas(gas),
+	})
+
+	i.handleGasRefund(res.GasRefund)
+	err := geth_vm.ErrExecutionReverted
+	if res.Success {
+		err = nil
+	}
+	return res.Output, uint64(res.GasLeft), err
```

Comment:
This code is similar to the `Call` implementation above and the static-call below. Can this be unified somehow?